src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -0,0 +1,206 @@
+  language: string;
+}
+
+export function OpenSavedQueryFlyout({
+  savedQueries,
+  onClose,
+  onQueryOpen,
+}: OpenSavedQueryFlyoutProps) {
+  // const [shouldRunOnOpen, setShouldRunOnOpen] = useState(false);
+  const [selectedQuery, setSelectedQuery] = useState<SavedQuery | undefined>(undefined);
+  const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
+  const [itemsPerPage, setItemsPerPage] = useState(10);
+  const pager = useRef(new Pager(savedQueries.length, itemsPerPage));
+  const [firstItemIndex, setFirstItemIndex] = useState(pager.current.getFirstItemIndex());
+  const [lastItemIndex, setLastItemIndex] = useState(pager.current.getLastItemIndex());
+
+  const onChange: EuiSearchBarProps['onChange'] = ({ query, error }) => {
+    if (!error) {
+      setSearchQuery(query);
+      pager.current.goToPageIndex(0);
+      setFirstItemIndex(pager.current.getFirstItemIndex());
+      setLastItemIndex(pager.current.getLastItemIndex());
+    }
+  };
+
+  const schema = {
+    strict: true,
+    fields: {
+      title: {
+        type: 'string',
+      },
+      description: {
+        type: 'string',
+      },
+      language: {
+        type: 'string',
+      },
+    },
+  };
+
+  const queryMap: {
+    [id: string]: { savedQuery: SavedQuery; savedQuerySearchableItem: SavedQuerySearchableItem };
+  } = {};
+  const queryLanguageSet = new Set<string>();
+  savedQueries.forEach((q) => {
+    queryLanguageSet.add(q.attributes.query.language);
+    queryMap[q.id] = {
+      savedQuery: q,
+      savedQuerySearchableItem: {
+        id: q.id,
+        title: q.attributes.title,
+        description: q.attributes.description,
+        language: q.attributes.query.language,
+      },
+    };
+  });
+
+  const filteredSavedQueries = EuiSearchBar.Query.execute(
+    searchQuery,
+    Object.values(queryMap).map((obj) => obj.savedQuerySearchableItem),
+    {
+      defaultFields: ['language', 'title', 'description'],
+    }
+  );
+
+  pager.current.setTotalItems(filteredSavedQueries.length);
+  const queriesOnCurrentPage = filteredSavedQueries.slice(firstItemIndex, lastItemIndex + 1);
+
+  return (
+    <EuiFlyout onClose={onClose} hideCloseButton>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h3>Saved queries</h3>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiSearchBar
+          box={{
+            placeholder: 'Search saved query',
+            incremental: true,
+            schema,
+          }}
+          filters={[
+            {
+              type: 'field_value_selection',
+              field: 'language',
+              name: 'Query language',
+              multiSelect: 'or',
+              options: Array.from(queryLanguageSet).map((language) => ({
+                value: language,
+                view: language.toUpperCase(),
+              })),
+            },
+          ]}
+          onChange={onChange}
+        />
+        <EuiSpacer />
```

Comment:
```suggestion
        <EuiSpacer size="s" />
```

@kgcreative can you please sign off on this suggestion?